Brevity#274 input of

ixmanuel opened the pull request:
Trial for #274 - InputOf

fabriciofx commented on src/main/java/org/cactoos/io/InputOf.java:
```diff
@@ -0,0 +1,201 @@
+     * @param src The file
+     */
+    public InputOf(final File src) {
+        this(new FileAsInput(src));
```
@ixmanuel You can't use lambda function here? I thought in something like:
```
public InputOf(final File src) {
  this(src -> new FileInputStream(src));
}
```

ixmanuel replied:
Yes, I tried according to the example that you exposed in the conversation # 312 but I didn't see more benefits for the rest of the cases, so I went back to the idea of Decorating Envelopes and thus, I didn't change anything, I have only created InputOf.

fabriciofx commented on src/main/java/org/cactoos/text/RegexText.java:
```diff
@@ -0,0 +1,78 @@
+/**
```
@ixmanuel WDYT to create a new PR only with Regex classes code?

ixmanuel replied:
Thanks. I've already fixed it.

fabriciofx commented on src/test/java/org/cactoos/text/RegexTextTest.java:
```diff
@@ -0,0 +1,58 @@
+/**
```
@ixmanuel The same above.

ixmanuel replied:
Thanks. I've already fixed it.

fabriciofx commented:
@ixmanuel Could you resubmit `RegexText` as a new issue/PR? Thanks in advance!

yegor256 commented:
@ixmanuel I don't like that `new InputOf(String)` would work as a `ResourceAsInput`. That's counter intuitive. I would expect it to do what `BytesAsInput` is doing now: convert a string content to the `Input`.

yegor256 commented:
@ixmanuel this is what I would expect:

  * `InputOf(File)`
  * `InputOf(Path)`
  * `InputOf(URL)`
  * `InputOf(URI)`
  * `InputOf(String)`
  * `InputOf(Text)`
  * `InputOf(Bytes)`
  * `InputOf(byte[])`
  * `InputOf(char[])`
  * `InputOf(StringBuilder)`
  * `InputOf(StringBuffer)`
  
maybe something else. "Resource" is not an option. We still need to have `ResourceAsInput`.

ixmanuel commented:
Yes, removing ResourceAsInput is counter intuitive.

ixmanuel commented:
@yegor256 InputOf done.

yegor256 commented on src/main/java/org/cactoos/io/InputOf.java:
```diff
@@ -0,0 +1,258 @@
+     *
+     * @param input The input
+     */
+    public InputOf(final Input input) {
```
@ixmanuel let's make it `private`

yegor256 commented:
@ixmanuel looks good, one small comment above

ixmanuel commented:
@yegor256, Now the main constructor is private.

yegor256 commented:
@rultor try to merge

rultor commented:
> @rultor try to merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/12444-316970545)

rultor commented:
> @rultor try to merge

@yegor256 Done! FYI, the full log is [here](http://www.rultor.com/t/12444-316970545) (took me 10min)

ixmanuel commented:
@yegor256, new test and fixed an independent ctor.

yegor256 commented:
@ixmanuel but the branch is merged already! Or I don't understand what's going on :)

ixmanuel commented:
@yegor256 I updated this almost at the same time you published it. Then I added some more improvements. Let me close this and open another one. Thanks.

ixmanuel commented:
@yegor256 Note: If you are reviewing now, BytesOf #340 and ReplacedAllText #341 are in synch with the branch you merged not with these post-updates. Now I'm creating a new pull request for theses post-updates of InputOf.

0crat commented:
Oops! Job `gh:yegor256/cactoos#312` was not in scope
